ArcadiaTeamsBot/Dialogs/MainDialog.cs

```diff
@@ -28,44 +28,22 @@ public MainDialog() : base(nameof(MainDialog))
             InitialDialogId = nameof(WaterfallDialog);
         }
 
-        private static async Task<DialogTurnResult> WelcomeStep(WaterfallStepContext stepContext, CancellationToken cancellationToken)
-        {
-            return await stepContext.PromptAsync(nameof(ConfirmPrompt),
-                new PromptOptions
-                {
-                    Prompt = MessageFactory.Text($"{WelcomeText}")
-                }, cancellationToken);
-        }
-
         private static async Task<DialogTurnResult> ChoiceStep(WaterfallStepContext stepContext, CancellationToken cancellationToken)
         {
-            if ((bool)stepContext.Result)
-            {
-                return await stepContext.PromptAsync(nameof(ChoicePrompt),
-                    new PromptOptions
-                    {
-                        Prompt = MessageFactory.Text("What would you like to do?"),
-                        Choices = ChoiceFactory.ToChoices(new List<string> { "New", "Current" }),
-                    }, cancellationToken);
-            }
-            else
-            {
-                await stepContext.Context.SendActivityAsync("Bye.", cancellationToken: cancellationToken);
-                return await stepContext.EndDialogAsync(cancellationToken: cancellationToken);
-            }
+            var reply = MessageFactory.Attachment(new List<Attachment>());
+            reply.Attachments.Add(Cards.GetChoiceCard().ToAttachment());
+            await stepContext.Context.SendActivityAsync(reply, cancellationToken);
+            return await stepContext.ContinueDialogAsync(cancellationToken);
         }
 
         private static async Task<DialogTurnResult> RequestStep(WaterfallStepContext stepContext, CancellationToken cancellationToken)
         {
-            stepContext.Values["choice"] = ((FoundChoice)stepContext.Result).Value;
-            if ((string)stepContext.Values["choice"] == "New")
+            stepContext.Values["choice"] = (string)stepContext.Result;
+            if ((string)stepContext.Values["choice"] == "New request")
```

Comment:
Shouldn't be a magic string, but should've depended on button value of appropriate card.

Reply:
Fixed